### PR TITLE
Use NuGet Central Package Management system

### LINKED ISSUE #8238: Use Directory.Packages.props to manage package versions (instead of Packages.targets)
*Directory.Packages.props* is related to the central package management system. We currently have a similar system of our design via `<PackageReference Update="" Version="">` nodes in a *Packages.targets* file. This issue is to move this information to a *Directory.Packages.props* instead of using our own system for managing versions in this repo.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #8238

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/9149)

### CODE REVIEW

File: eng/imports/Packages.targets [hunk 1]
```diff
@@ -1,129 +0,0 @@
...
-    <PackageReference Update="Microsoft.VisualStudio.Utilities"                                       Version="17.7.36307-preview.2" />
-    <PackageReference Update="Microsoft.VisualStudio.Validation"                                      Version="17.6.11" />
-    <PackageReference Update="Microsoft.VisualStudio.XmlEditor"                                       Version="17.3.32804.24" />
-    <PackageReference Update="Microsoft.VSSDK.BuildTools"                                             Version="17.5.18-preview1" />
-    <PackageReference Update="System.Threading.Tasks.Dataflow"                                        Version="7.0.0" />
-    <PackageReference Update="Microsoft.VSDesigner"                                                   Version="17.0.0-preview-2-31223-026" />
-    <PackageReference Update="VsWebSite.Interop"                                                      Version="16.8.30523.219" />
-
-    <!-- https://github.com/dotnet/roslyn/issues/53877 -->
-    <PackageReference Include="StreamJsonRpc"                                                         Version="2.16.21-preview" />
```
Comment (drewnoakes) [inline eng/imports/Packages.targets:67]: I"m confused about this part of the PR. This is an `Include` among a bunch of `Update`s. I'd expect us to need a new `Include` somewhere else in the repo for this to build.

It's also possible that the reason we needed to add an explicit package reference to `StreamJsonRpc` has been fixed, and that we can remove it altogether.

File: Directory.Packages.props [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    <PackageVersion Include="Microsoft.VisualStudio.Utilities"                                       Version="17.7.36307-preview.2" />
+    <PackageVersion Include="Microsoft.VisualStudio.Validation"                                      Version="17.6.11" />
+    <PackageVersion Include="Microsoft.VisualStudio.XmlEditor"                                       Version="17.3.32804.24" />
+    <PackageVersion Include="Microsoft.VSSDK.BuildTools"                                             Version="17.5.18-preview1" />
+    <PackageVersion Include="System.Threading.Tasks.Dataflow"                                        Version="7.0.0" />
+    <PackageVersion Include="Microsoft.VSDesigner"                                                   Version="17.0.0-preview-2-31223-026" />
+    <PackageVersion Include="VsWebSite.Interop"                                                      Version="16.8.30523.219" />
+
+    <!-- https://github.com/dotnet/roslyn/issues/53877 -->
+    <PackageVersion Include="StreamJsonRpc"                                                         Version="2.16.21-preview" />
```
Comment (drewnoakes) [inline Directory.Packages.props:70]: If this is not needed any more it can be removed.

File: Directory.Packages.props [hunk 3]
```diff
@@ -0,0 +1,131 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information. -->
+<Project>
+
+  <Import Project="eng\imports\Versions.props" />
+
+  <PropertyGroup>
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+  </PropertyGroup>
+  <ItemGroup>
```
Comment (drewnoakes) [inline Directory.Packages.props:9]: nit: a blank line here would be nice

**Review by drewnoakes — APPROVED** [review]
LGTM. I've wanted us to dogfood this feature for a while. Thanks for making it happen.

File: Directory.Packages.props (hunk 2 above)
Comment (smitpatel) [reply Directory.Packages.props]: Local build passed, let's see if CI is happy with it.

**Pull request merged**